intranet/apps/eighth/management/commands/dev_generate_signups.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+import random
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand, CommandError
+from django.utils.timezone import localdate, make_aware
+
+from intranet.apps.eighth.models import EighthBlock, EighthScheduledActivity, EighthSignup
+
+
+class Command(BaseCommand):
+    help = "Generate 8th Period Signups from existing users and activities"
+
+    def add_arguments(self, parser):
+        # Positional arguments
+        parser.add_argument("date")
+
+    def handle(self, *args, **kwargs):
+        try:
+            ablock = EighthBlock.objects.get(date=localdate(make_aware(datetime.datetime.strptime(kwargs["date"], "%m/%d/%Y"))), block_letter="A")
+            bblock = EighthBlock.objects.get(date=localdate(make_aware(datetime.datetime.strptime(kwargs["date"], "%m/%d/%Y"))), block_letter="B")
+        except EighthBlock.DoesNotExist as ex:
+            raise CommandError(str(ex)) from ex
+
+        a_activities = EighthScheduledActivity.objects.filter(block=ablock)
+        b_activities = EighthScheduledActivity.objects.filter(block=bblock)
+        students = get_user_model().objects.filter(user_type="student")
+
+        for student in students:
+            a_activity = random.choice(a_activities)
```

Comment:
This doesn't seem to take into account activities that are restricted.

Reply:
`random.choice` is expensive and not the optimal way to do this, especially if the size of `a_activities` is large and the records are not hot in the heap.
Pulling a given index with `random.randint` would be better.